lessons/guesser.rb

```diff
@@ -0,0 +1,77 @@
+def valid_answer?(input)
+    valid = ["too low", "too high", "correct"]
+    valid.include?(input.downcase)
+end
+
+def valid_yes_no?(input)
+    input.downcase == "yes" || input.downcase == "no"
+end
+
+def guessing_game
+    loop do
+        puts "Please think of a number from 1 to 100"
+        
+        low = 1
+        high = 100
+        guess = nil
+        lying = false
+
+        loop do 
+            #binary search
+            guess = (low + high) / 2
+            puts "Your number is #{guess}. Enter 'correct', too low', or 'too high'"
+            answer = gets.chomp.downcase
+
+            unless valid_answer?(answer)
+                puts "Invalid input. Enter 'correct', too low', or 'too high'"
+                next
+            end
+
+            if answer == "correct"
+                break
+            elsif answer == 'too low'
+                low = guess + 1
+            elsif answer == 'too high' 
+                high = guess - 1
+            end
+
+            #not sure if I need this if statement. looks the code will never get to this point
+            if high < low
+                lying = true
+                break
+            end
+        end
```

Comment:
This is the right way to say that the user is lying, IMO. See comment on lines 55/56